### PR TITLE
fix(content-uploader): make disabled buttons unfocusable

### PR DESCRIPTION
**Pull request by QingyuChai**
Similar fix to this: https://github.com/box/box-ui-elements/pull/3243/files

Essentially: The `isDisabled` prop does not set the `disabled` property, which means that it is still focusable by the keyboard.

Potential idea to have `isDisabled` set the `disabled` property as well, but because it might cause unintended consequences in parent apps we won't do so here.

https://www.w3.org/WAI/ARIA/apg/patterns/button/

image

### CODE REVIEW

File: src/elements/content-uploader/Footer.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ type Props = {
 };
 
 const Footer = ({ isLoading, hasFiles, errorCode, onCancel, onClose, onUpload, fileLimit, isDone }: Props) => {
+    const isCloseButtonDisabled = hasFiles;
+    const isCancelButtonDisabled = !hasFiles || isDone;
+    const isUploadButtonDisabled = !hasFiles;
```
Comment (QingyuChai) [inline src/elements/content-uploader/Footer.js:28]: moved the closeButton and uploadButton disabled booleans up as well, for readability. Not necessary though

**Review by tjuanitas — APPROVED** [review]

**Review by megansmith-box — APPROVED** [review]

File: src/elements/content-uploader/Footer.js (hunk 1 above)
Comment (jpan-box) [reply src/elements/content-uploader/Footer.js:28]: i dig it

**Review by jpan-box — APPROVED** [review]

File: src/elements/content-uploader/Footer.js [hunk 2]
```diff
@@ -36,17 +40,33 @@ const Footer = ({ isLoading, hasFiles, errorCode, onCancel, onClose, onUpload, f
         <div className="bcu-footer">
             <div className="bcu-footer-left">
                 {onClose ? (
-                    <Button isDisabled={hasFiles} onClick={onClose} type="button">
+                    <Button
+                        disabled={isCloseButtonDisabled}
```
Comment (JChan106) [inline src/elements/content-uploader/Footer.js:44]: what's the difference between disabled and isDisabled?

File: src/elements/content-uploader/__tests__/Footer.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import noop from 'lodash/noop';
```
Comment (JChan106) [inline src/elements/content-uploader/__tests__/Footer.test.js:3]: move lodash before enzyme

File: src/elements/content-uploader/__tests__/Footer.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import Button from '../../../components/button/Button';
+import PrimaryButton from '../../../components/primary-button/PrimaryButton';
+import Footer from '../Footer';
```
Comment (JChan106) [inline src/elements/content-uploader/__tests__/Footer.test.js:6]: move footer before primary button

File: src/elements/content-uploader/Footer.js (hunk 2 above)
Comment (QingyuChai) [reply src/elements/content-uploader/Footer.js:44]: isDisabled sets the styles/classes for the button, while disabled is passed onto the html element <button />, which prevents focus on the element

File: src/elements/content-uploader/__tests__/Footer.test.js (hunk 3 above)
Comment (QingyuChai) [reply src/elements/content-uploader/__tests__/Footer.test.js:3]: I think we try to follow a convention where react packages are at the top, and we alphasort the rest

**Review by JChan106 — APPROVED** [review]
lgtm

**Review by tjuanitas — APPROVED** [review]

**Pull request merged**